interactions/api/models/webhook.py

```diff
@@ -0,0 +1,301 @@
+        :return: The modified webhook object
+        :rtype: Webhook
+        """
+
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+
+        if channel_id in (None, MISSING) and not self.token:
+            raise ValueError("no token was found, please specify a channel id!")
+
+        payload = {}
+
+        if name is not MISSING:
+            payload["name"] = name
+
+        if avatar is not MISSING:
+            payload["avatar"] = avatar.data
+
+        if channel_id is not MISSING:
+            payload["channel_id"] = channel_id
+
+        res = await self._client.modify_webhook(
+            webhook_id=int(self.id),
+            payload=payload,
+            webhook_token=None if channel_id else self.token,
+        )
+
+        webhook = Webhook(**res, _client=self._client)
+
+        for attr in self.__slots__:
+            setattr(self, attr, getattr(webhook, attr))
+
+        return webhook
+
+    async def execute(
+        self,
+        content: Optional[str] = MISSING,
+        username: Optional[str] = MISSING,
+        avatar_url: Optional[str] = MISSING,
+        tts: Optional[bool] = MISSING,
+        embeds: Optional[Union["Embed", List["Embed"]]] = MISSING,  # noqa
+        allowed_mentions: Any = MISSING,
+        components: Optional[
+            Union[
+                "ActionRow",  # noqa
+                "Button",  # noqa
+                "SelectMenu",  # noqa
+                List["ActionRow"],  # noqa
+                List["Button"],  # noqa
+                List["SelectMenu"],  # noqa
+            ]
+        ] = MISSING,
+        files: Optional[Union[File, List[File]]] = MISSING,
+        thread_id: Optional[int] = MISSING,
+    ) -> Optional["Message"]:  # noqa
+        """
+        Executes the webhook. All parameters to this function are optional.
+
+        .. important::
+            The ``components`` argument requires an application-owned webhook.
+
+        :param content: the message contents (up to 2000 characters)
+        :type content: str
+        :param username: override the default username of the webhook
+        :type username: str
+        :param avatar_url: override the default avatar of the webhook
+        :type avatar_url: str
+        :param tts: true if this is a TTS message
+        :type tts: bool
+        :param embeds: embedded ``rich`` content
+        :type embeds: Union[Embed, List[Embed]]
+        :param allowed_mentions: allowed mentions for the message
+        :type allowed_mentions: dict
+        :param components: the components to include with the message
+        :type components: Union[ActionRow, Button, SelectMenu, List[ActionRow], List[Button], List[SelectMenu]]
+        :param files: The files to attach to the message
+        :type files: Union[File, List[File]]
+        :param thread_id: Send a message to a specified thread within a webhook's channel. The thread will automatically be unarchived
+        :type thread_id: int
```

Comment:
docs moment: what `execute` should return?

Reply:
Optional[Message], i'm fixing it